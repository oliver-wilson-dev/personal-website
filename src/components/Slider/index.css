.tileWrapper {
  max-height: none;
  display: flex;
  overflow: hidden;
  justify-content: center;
  align-items: flex-start;
}

.slider {
  display: flex;
  overflow: scroll;
  max-width: 50%;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 10px 200px;
  scroll-snap-type: x mandatory;

  &::-webkit-scrollbar {
    display: none;
    -ms-overflow-style: none; /* IE and Edge */
    scrollbar-width: none; /* Firefox */
  }

  position: relative;
  &::after {
    position: absolute;
    content: "";
    right: -200vw;
    width: 5vw;
    height: 1px;
  }
}

@media (min-width: 0px) and (max-width: 767px) {
  .tileWrapper {
    max-width: 100vw;
  }
}

@media (min-width: 1024px) {
  .slider {
    max-width: 100%;
  }
}
