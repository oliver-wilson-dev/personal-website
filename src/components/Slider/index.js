/* eslint-disable react/prop-types */
import React from 'react';
import cn from 'classnames';
import styles from './index.css';

const Slider = ({ children }) => (
  <div className={cn(styles.tileWrapper)}>
    <div className={cn(styles.slider)}>
      {children}
    </div>
  </div>
);

export default Slider;
