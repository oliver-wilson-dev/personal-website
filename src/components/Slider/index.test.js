import React from 'react';
import { shallow } from 'enzyme';
import Slider from './index';

const headingText = 'hello world';

const render = () => shallow(<Slider>{headingText}</Slider>);

describe('Slider', () => {
  it('should render correctly', () => {
    const component = render();

    expect(component).toMatchSnapshot();
  });
});
