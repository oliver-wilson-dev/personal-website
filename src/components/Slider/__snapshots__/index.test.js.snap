// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly 1`] = `
<div
  className="tileWrapper"
>
  <div
    className="slider"
  >
    hello world
  </div>
</div>
`;
